pkg/asset/installconfig/aws/basedomain.go

```diff
@@ -91,7 +93,17 @@ func GetPublicZone(name string) (*route53.HostedZone, error) {
 	if err != nil {
 		return nil, errors.Wrap(err, "getting AWS session")
 	}
-	client := route53.New(session)
+	// TODO Remove the hard code api endpoint after it's GA in AWS China
+	// If it's AWS China region, we have to set region to "cn-northwest-1"
+	// as route53 is hosted in that region.
+	awsConfig := aws.NewConfig()
+	switch *session.Config.Region {
```

Comment:
this seems odd. GetSession doesn't set this value.

Reply:
@abhinavdahiya This does exist when you specify the "AWS_REGION" or "AWS__DEFAULT_REGION" env or have the region in ~/.aws/config, I think that's why I didn't see any errors during my testing, and yes,  I think we should go with the "region" in the install-config, not the aws credential config.